### ISSUE TITLE
Literal subset of enum doesn't preserve exhaustiveness

### ISSUE THREAD

**Issue by anuraaga** [astral-sh/ty#3531]
### Summary

This started happening in ty 0.0.38 and continues in 0.0.39. It was fine in 0.0.37.

A match on a `Literal` subset of enum values does not recognize exhaustiveness causing `assert_never` to fail. It seems to use the full enum definition instead of the subset. Note a `TypeAlias` does not seem to improve it.

```python
from __future__ import annotations

from enum import IntEnum
from typing import Literal, assert_never


class E(IntEnum):
    A = 1
    B = 2
    C = 3
    D = 4  # member outside the Literal


def f(v: Literal[E.A, E.B, E.C]) -> None:
    match v:
        case E.A | E.B:  # OR pattern — triggers the bug
            pass
        case E.C:
            pass
        case _:
            assert_never(v)  # error: Expected Never, found Literal[E.A, E.B, E.C]
```

```
❯ uvx ty==0.0.38 check --python-version 3.11 repro.py
Installed 1 package in 19ms
error[[type-assertion-failure](https://ty.dev/rules#type-assertion-failure)]: Argument does not have asserted type `Never`
  --> repro.py:21:13
   |
21 |             assert_never(v)  # error: Expected Never, found Literal[E.A, E.B, E.C]
   |             ^^^^^^^^^^^^^-^
   |                          |
   |                          Inferred type of argument is `Literal[E.A, E.B, E.C]`
   |
info: `Never` and `Literal[E.A, E.B, E.C]` are not equivalent types

Found 1 diagnostic
```

Presumably it's an unintended side effect of https://github.com/astral-sh/ruff/pull/24961

### Version

ty 0.0.39 (020512517 2026-05-22)

**Comment by anuraaga** [comment]
Realized the issue isn't about the enum subset but just resolving the `|` cases. I sent https://github.com/astral-sh/ruff/pull/25379 with a proposed fix.
